ksql-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -89,7 +89,7 @@ public JoinNode(
             ? left.getKeyField()
             : KeyField.of(leftKeyCol.ref());
 
-    this.schema = buildSchema(left, right);
+    this.schema = JoinParamsFactory.createSchema(left.getSchema(), right.getSchema());
```

Comment:
This is just bridge code. Planning to drop this in a follow-up. We shouldn't need to get the schema from the logical plan nodes.